packages/vite/src/node/ssr/runtime/source-map/decoder.ts

```diff
@@ -0,0 +1,260 @@
+import { posixResolve } from '../utils'
+
+interface SourceMapLike {
+  version: number
+  mappings?: string
+  names?: string[]
+  sources?: string[]
+  sourcesContent?: string[]
+}
+
+type OriginalMapping = {
+  source: string | null
+  line: number
+  column: number
+  name: string | null
+}
+
+type Needle = {
+  line: number
+  column: number
+}
+
+export class DecodedMap {
+  _encoded: string
+  _decoded: undefined | number[][][]
+  _decodedMemo: Stats
+  url: string
+  version: number
+  names: string[] = []
+  resolvedSources: string[]
+
+  constructor(
+    public map: SourceMapLike,
+    from: string,
+  ) {
+    const { mappings, names, sources } = map
+    this.version = map.version
+    this.names = names || []
+    this._encoded = mappings || ''
+    this._decodedMemo = memoizedState()
+    this.url = from
+    this.resolvedSources = (sources || []).map((s) =>
+      posixResolve(s || '', from),
+    )
+  }
+}
+
+// This is a copy of all methods that we need for decoding a source map from "@jridgewell/trace-mapping"
```

Comment:
I am trying to avoid any bundling for the runtime part. The code for `@jridgewell/trace-mapping` is shared here and in other places. Rollup creates a chunk that can contain a lot of unused code in the runtime, and also injects `__require` via our plugin (I added an exception for a runtime chunk to not do that because we cannot assume it's running in Node). Another reason I inline it here is because the code elimination is bad, we just need a very small part of this package.

Reply:
As discussed on Discord, let's leave it as is then and polish out the bundling part later. We can maybe look into a specific Rollup build for the runtime so we can optimize its output that doesn't contain the unnecessary stuff.